lib/core/src/Cardano/Wallet/Api/Types.hs

```diff
@@ -1043,11 +1046,21 @@ data ApiWalletSignData = ApiWalletSignData
     } deriving (Eq, Generic, Show)
       deriving anyclass NFData
 
-newtype ApiVerificationKey = ApiVerificationKey
+newtype ApiVerificationKeyShelley = ApiVerificationKeyShelley
     { getApiVerificationKey :: (ByteString, Role)
     } deriving (Eq, Generic, Show)
       deriving anyclass NFData
 
+data VerificationKeyHashing = HashingApplied | WithoutHashing
+    deriving (Eq, Generic, Show)
+    deriving anyclass NFData
+
+data ApiVerificationKeyShared = ApiVerificationKeyShared
+    { getApiVerificationKey :: (ByteString, Role)
+    , hashed :: VerificationKeyHashing
+    } deriving (Eq, Generic, Show)
+      deriving anyclass NFData
+
 data ApiPostAccountKeyData = ApiPostAccountKeyData
     { passphrase :: ApiT (Passphrase "raw")
     , extended :: Bool
```

Comment:
Only seeing this now, and I think we can do better than `Bool` here.

Reply:
introduced ```data KeyFormat = Extended | NonExtended ``` as a replacement